backend/src/manager/mod.rs

```diff
@@ -510,12 +513,103 @@ async fn manager_thread_loop(
     }
 }
 
+struct LongRunningHandle(NonDetachingJoinHandle<()>);
+pub struct CommandInserter {
+    command_counter: Mutex<u64>,
+    input: UnboundedSender<InputJsonRpc>,
+    outputs: Arc<Mutex<BTreeMap<u64, UnboundedSender<embassy_container_init::Output>>>>,
+}
+impl Drop for CommandInserter {
+    fn drop(&mut self) {
+        use embassy_container_init::{Input, JsonRpc, RpcId};
+        for i in 0..*self.command_counter.blocking_lock() {
+            let _ignored_result = self.input.send(JsonRpc::new(RpcId::UInt(i), Input::Term()));
+        }
+    }
+}
+impl CommandInserter {
+    fn new(
+        long_running: LongRunning,
+        input: UnboundedSender<InputJsonRpc>,
+    ) -> (Self, LongRunningHandle) {
+        let LongRunning {
+            mut output,
+            running_output,
+        } = long_running;
+        let command_counter = Mutex::new(0);
+        let outputs: Arc<Mutex<BTreeMap<u64, UnboundedSender<embassy_container_init::Output>>>> =
+            Default::default();
+        let handle = LongRunningHandle(running_output);
+        tokio::spawn({
+            let outputs = outputs.clone();
+            async move {
+                while let Some(output) = output.recv().await {
```

Comment:
:notebook: This will stay open as long as there is output...
Which is staying open as long as there is input. 
Which is controlled by the the input owned by this struct.